tests/functional/test_contract_event.py

```diff
@@ -0,0 +1,231 @@
+    assert_log_values(logs[1], 1)
+
+
+def test_contract_logs_range_start_and_stop(contract_instance, owner, chain):
+    # Create 1 event
+    contract_instance.setNumber(1, sender=owner)
+
+    # Grab start block after first event
+    start_block = chain.blocks.height
+
+    contract_instance.setNumber(2, sender=owner)
+    contract_instance.setNumber(3, sender=owner)
+
+    stop = 30  # Stop can be bigger than height, it doesn't not matter
+    logs = [log for log in contract_instance.NumberChange.range(start_block, stop=stop)]
+    assert len(logs) == 3, "Unexpected number of logs"
+
+
+def test_contract_logs_range_only_stop(contract_instance, owner, chain):
+    # Create 1 event
+    contract_instance.setNumber(1, sender=owner)
+    contract_instance.setNumber(2, sender=owner)
+    contract_instance.setNumber(3, sender=owner)
+
+    stop = 100  # Stop can be bigger than height, it doesn't not matter
+    logs = [log for log in contract_instance.NumberChange.range(stop)]
+    assert len(logs) == 3, "Unexpected number of logs"
+
+
+def test_contract_logs_range_with_paging(contract_instance, owner, chain, assert_log_values):
+    # Create 1 log each in the first 3 blocks.
+    for i in range(3):
+        contract_instance.setNumber(i + 1, sender=owner)
+
+    # Mine 3 times to ensure we can handle uneventful blocks.
+    for i in range(3):
+        chain.mine()
+
+    # Create one more log after the empty blocks.
+    contract_instance.setNumber(100, sender=owner)
+
+    logs = [log for log in contract_instance.NumberChange.range(100, block_page_size=1)]
+    assert len(logs) == 4, "Unexpected number of logs"
+    assert_log_values(logs[0], 1)
+    assert_log_values(logs[1], 2)
+    assert_log_values(logs[2], 3)
+    assert_log_values(logs[3], 100, previous_number=3)
+
+
+def test_contract_logs_range_over_paging(contract_instance, owner, chain):
+    # Create 1 log each in the first 3 blocks.
+    for i in range(3):
+        contract_instance.setNumber(i + 1, sender=owner)
+
+    # 50 is way more than 3 but it shouldn't matter.
+    logs = [log for log in contract_instance.NumberChange.range(100, block_page_size=50)]
+    assert len(logs) == 3, "Unexpected number of logs"
+
+
+def test_contract_logs_from_non_indexed_range(contract_instance, owner):
+    contract_instance.setNumber(1, sender=owner)
+    with pytest.raises(DecodingError):
+        _ = [
+            log for log in contract_instance.NumberChange.range(0, event_parameters={"prevNum": 1})
+        ]
+
+
+def test_poll_logs_stop_block_not_in_future(
+    chain_at_block_5, vyper_contract_instance, eth_tester_provider
+):
+    bad_stop_block = chain_at_block_5.blocks.height
+
+    with pytest.raises(ValueError) as err:
+        _ = [x for x in vyper_contract_instance.NumberChange.poll_logs(stop_block=bad_stop_block)]
+
+    assert str(err.value) == "'stop' argument must be in the future."
+
+
+def test_poll_logs(vyper_contract_instance, eth_tester_provider, owner, poll_daemon):
```

Comment:
all other tests in this file were simply moved over **Except** for this one... this is the new test!